src/parser/SparqlParser.cpp

```diff
@@ -19,16 +19,16 @@ SparqlParser::SparqlParser(const string& query) : lexer_(query), query_(query) {
   LOG(DEBUG) << "Parsing " << query << std::endl;
 }
 
-// _____________________________________________________________________________
-ParsedQuery SparqlParser::parse() {
-  ParsedQuery result = parseWithAntlr(
-      &AntlrParser::query,
-      {{INTERNAL_PREDICATE_PREFIX_NAME, INTERNAL_PREDICATE_PREFIX_IRI}},
-      query_);
-
-  lexer_.expectEmpty();
-
-  return result;
+ParsedQuery SparqlParser::parseQuery(std::string_view query) {
+  sparqlParserHelpers::ParserAndVisitor p{
+      query, {{INTERNAL_PREDICATE_PREFIX_NAME, INTERNAL_PREDICATE_PREFIX_IRI}}};
+  auto resultOfParseAndRemainingText = p.parseTypesafe(&AntlrParser::query);
+  if (!resultOfParseAndRemainingText.remainingText_.empty()) {
+    // TODO: add Exception Metadata
+    throw ParseException("Query couldn't be parsed completely. Trailing: " +
+                         resultOfParseAndRemainingText.remainingText_);
+  }
```

Comment:
Maybe `"A query was succesfully parsed, but the following remainder of the input was ignored by the parsed: <remainingText>"`.
And the remaining text should be as you stated denoted by the ExceptionMetaData.